tests/components/zha/test_sensor.py

```diff
@@ -265,3 +265,58 @@ async def test_temp_uom(
     assert state is not None
     assert round(float(state.state)) == expected
     assert state.attributes[ATTR_UNIT_OF_MEASUREMENT] == uom
+
+
+async def test_electrical_measurement_init(
+    hass, zigpy_device_mock, zha_device_joined,
+):
+    """Test proper initialization of the electrical measurement cluster."""
+
+    cluster_id = homeautomation.ElectricalMeasurement.cluster_id
+    zigpy_device = zigpy_device_mock(
+        {
+            1: {
+                "in_clusters": [cluster_id, general.Basic.cluster_id],
+                "out_cluster": [],
+                "device_type": 0x0000,
+            }
+        }
+    )
+    cluster = zigpy_device.endpoints[1].in_clusters[cluster_id]
+    zha_device = await zha_device_joined(zigpy_device)
+    entity_id = await find_entity_id(DOMAIN, zha_device, hass)
+
+    # allow traffic to flow through the gateway and devices
+    await async_enable_traffic(hass, [zha_device])
+
+    # test that the sensor now have a state of unknown
+    assert hass.states.get(entity_id).state == STATE_UNKNOWN
+
+    await send_attributes_report(hass, cluster, {0: 1, 1291: 100, 10: 1000})
+    assert int(hass.states.get(entity_id).state) == 100
+
+    channel = zha_device.channels.pools[0].all_channels["1:0x0b04"]
+    assert channel.divisor == 1
+    assert channel.multiplier == 1
+
+    # update power divisor
+    await send_attributes_report(hass, cluster, {0: 1, 1291: 20, 0x0403: 5, 10: 1000})
+    assert channel.divisor == 5
+    assert channel.multiplier == 1
+    assert hass.states.get(entity_id).state == "4.0"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5844602)